scripts/lib/nginx-prune-cache-subscript.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/local/bin/bash
+
+# This subscript is expected to be run inside docker container using 'bash'
+# image. The image is based on Alpine Linux. It's tools (find, stat, awk, sort)
+# are non-standard versions from BusyBox.
+
+MAX_CACHE_DIR_SIZE=20000000000
+MAX_KEEP_FILE_SIZE=1000000000
+
+total=0
+
+# We sort files by time, newest files are first. Format is:
+# time (last modification as seconds since Epoch), filepath, size (bytes)
+find /home/user/skynet-webportal/docker/data/nginx/cache -type f -exec stat -c "%Y %n %s" {} + | sort -rgk1 | while read line
+do
+    size=$(echo $line | awk '{print $3}')
+    new_total=$(($total + $size))
+
+    # We always delete all files larger than MAX_KEEP_FILE_SIZE.
+    # We keep all files smaller than MAX_KEEP_FILE_SIZE when cache size is
+    # below MAX_CACHE_DIR_SIZE, then we delete also smaller files.
+    if (("$size" <= "$MAX_KEEP_FILE_SIZE" && "$new_total" < "$MAX_CACHE_DIR_SIZE"))
```

Comment:
I think this is still wrong. 
new total is incrementing, which means it will always be less than the dir size until we have iterated over enough of the directory. 

I think the logic should be 
```
if cache < max_size
    end

if file > max_size
    delete file
    cache = cache - file
```

Reply:
Splitting it up to improve readability:

```
if total < max_size && file > max_size
    delete file
    total -= file size
    
if total < max_size
    total += file size
else
    delete file
```

Reply:
when we have `MAX_KEEP_FILE_SIZE = 1GB`
and there are 10,000x 500MB files - then nothing is pruned from cache

Reply:
The logic in PR is:
```
list all files, newest first
if (file is small) && (we have room in cache)
    keep file
    continue
delete file
```

Reply:
I see now. got it,